### ISSUE TITLE
[CI/CD] Azure Pipeline: issue with `_sqlite3` module

### ISSUE THREAD

**Issue by picnixz** [python/cpython#120159]
# Bug report

### Bug description:

This is something I wanted to point out but I'm not sure whether it's a known bug or not (if so, feel free to close the issue). I couldn't find a related issue.

---

The Azure Pipeline is failing on Windows, similarly to https://github.com/python/cpython/issues/84395 but with a more explicit error. There are two failing tests, namely

- `test.test_ctypes.test_loading.LoaderTest.test_load_dll_with_flags`
- `test.test_import.ImportTests.test_dll_dependency_import`

Both of them fail because they cannot find the DLL related to the `sqlite3` module. The [first failure](https://dev.azure.com/Python/cpython/_build/results?buildId=153961&view=logs&j=0fcf9c9b-89fc-526f-8708-363e467e119e&t=fa5ef4ee-3911-591e-4444-19482ab189b7&l=945) failure is actually very clear:

```text
 Traceback (most recent call last):
  File "D:\a\1\b\layout-appx-win32\Lib\test\test_ctypes\test_loading.py", line 157, in test_load_dll_with_flags
    shutil.copy(os.path.join(os.path.dirname(src), "sqlite3" + ext),
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                os.path.join(tmp, "sqlite3" + ext))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\a\1\b\layout-appx-win32\Lib\shutil.py", line 428, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
    ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\a\1\b\layout-appx-win32\Lib\shutil.py", line 260, in copyfile
    with open(src, 'rb') as fsrc:
         ~~~~^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\a\\1\\b\\layout-appx-win32\\DLLs\\sqlite3.dll'
```

I think the issue is with the DLLs directory which is never created (AFAICT), as illustrated by the current code:

https://github.com/python/cpython/blob/fd104dfcb838d735ef8128e3539d7a730d403422/Lib/test/test_ctypes/test_loading.py#L150-L158

There should be some

```python
tmp2 = os.path.join(tmp, "DLLs")
os.mkdir(tmp2)
```

similar to what is being done in `test.test_import.ImportTests.test_dll_dependency_import`. Note that the source DLL should be present since it is [generated](https://dev.azure.com/Python/cpython/_build/results?buildId=153961&view=logs&j=0fcf9c9b-89fc-526f-8708-363e467e119e&t=0e014acc-15f7-5613-9f10-e24e9946e9cc&l=796) during the build.

The [second failure](https://dev.azure.com/Python/cpython/_build/results?buildId=153961&view=logs&j=0fcf9c9b-89fc-526f-8708-363e467e119e&t=fa5ef4ee-3911-591e-4444-19482ab189b7&l=976
) occurs with:

```text
ERROR: test_dll_dependency_import (test.test_import.ImportTests.test_dll_dependency_import)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\1\b\layout-appx-win32\Lib\test\test_import\__init__.py", line 762, in test_dll_dependency_import
...
subprocess.CalledProcessError: Command '[
	'C:\\Users\\VssAdministrator\\AppData\\Local\\Temp\\test_python_w9okac52\\tmp7ps_2s1g\\python.exe', 
	'-Sc', 
	"import os;p = os.add_dll_directory('D:\\\\a\\\\1\\\\b\\\\layout-appx-win32\\\\DLLs');import _sqlite3;p.close"
]' returned non-zero exit status 1.
```

Here I think what occurs is that the files are not copied correctly (the DLLs directory is correctly created but I assume that the issue is due to the missing dll file, or perhaps it's moved incorrectly). Here is the code setting up the environment:

https://github.com/python/cpython/blob/fd104dfcb838d735ef8128e3539d7a730d403422/Lib/test/test_import/__init__.py#L738-L755

### CPython versions tested on:

CPython main branch

### Operating systems tested on:

Other

**Comment by eryksun** [comment]
@zooba, this issue with finding "sqlite3.dll" in the appx layout is resolved by #120133, right?

**Comment by zooba** [comment]
> this issue with finding "sqlite3.dll" in the appx layout is resolved by #120133, right?

Yeah, it will. I noticed the error because my own test builds were failing the other day, but didn't bother copying the whole thing into the bug this time (serves me right...)
